crates/forge_services/src/tool_services/fs_search.rs

```diff
@@ -426,12 +445,10 @@ mod test {
             .await
             .unwrap();
 
-        // Should find matches in both files now (binary file converted with lossy)
+        // Should only find matches in the text file (binary file skipped)
         assert!(actual.is_some());
         let result = actual.unwrap();
-        assert_eq!(result.matches.len(), 2);
-        let paths: HashSet<_> = result.matches.iter().map(|m| &m.path).collect();
-        assert!(paths.iter().any(|p| p.ends_with("valid.txt")));
-        assert!(paths.iter().any(|p| p.ends_with("binary.bin")));
+        assert_eq!(result.matches.len(), 1);
+        assert!(result.matches[0].path.ends_with("valid.txt"));
```

Comment:
Add a test to search for binary files. The tool allows searching for files and also searching for content inside files. Add a test to make sure that a bin file can be searched using a tool.